report_substitute/wizards/mail_compose_message.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class MailComposeMessage(models.TransientModel):
+
+    _inherit = 'mail.compose.message'
+
+    @api.multi
+    @api.onchange('template_id')
+    def onchange_template_id_wrapper(self):
+        old_report_template = False
+        if (
+            self.template_id
+            and self.template_id.report_template
+            and self.env.context.get('active_ids')
```

Comment:
... and `self.template_id.report_template.action_report_substitution_rule_ids` ?